src/main/java/com/iexec/worker/feign/CustomCoreFeignClient.java

```diff
@@ -129,21 +129,21 @@ public Optional<ReplicateTaskSummary> getAvailableReplicateTaskSummary(long last
     }
 
     public TaskNotificationType updateReplicateStatus(String chainTaskId, ReplicateStatusUpdate replicateStatusUpdate) {
-        Map<String, Object> arguments = new HashMap<>();
-        arguments.put(JWTOKEN, loginService.getToken());
-        arguments.put("chainTaskId", chainTaskId);
-        arguments.put("statusUpdate", replicateStatusUpdate);
-
-        HttpCall<TaskNotificationType> httpCall = args ->
-                coreClient.updateReplicateStatus((String) args.get(JWTOKEN), (String) args.get("chainTaskId"),
-                        (ReplicateStatusUpdate) args.get("statusUpdate"));
+        final ResponseEntity<TaskNotificationType> response = coreClient.updateReplicateStatus(
+                loginService.getToken(),
+                chainTaskId,
+                replicateStatusUpdate
+        );
+        if (response.getStatusCode() == HttpStatus.UNAUTHORIZED) {
+            login();
+        }
 
-        ResponseEntity<TaskNotificationType> response = makeHttpCall(httpCall, arguments, "updateReplicateStatus");
-        if (!is2xxSuccess(response)) {
+        if (!response.getStatusCode().is2xxSuccessful()) {
             return null;
         }
 
+        final TaskNotificationType notificationType = response.getBody();
         log.info(replicateStatusUpdate.getStatus().toString() + " [chainTaskId:{}]", chainTaskId);
```

Comment:
Should we move those 2 lines before the `is2xx` check and add the status code to the logging event ?

Reply:
Please see https://github.com/iExecBlockchainComputing/iexec-worker/pull/499/commits/0916b53c5eea7657e338d429f2bf055696bc2bdd.